auth_proxy/services/oauth.py

```diff
@@ -214,11 +214,16 @@ def cb_tokensetter(self, token, request, *args, **kwargs):
         assert request.user is not None
 
         today = arrow.now().datetime
-        old = self.db.session.query(Token).\
+        old_tokens = self.db.session.query(Token).\
             filter_by(client_id=request.client.client_id).\
             filter(Token.approval_expires >= today).\
-            one()
-        new = old.refresh(**token)
+            order_by(Token.approval_expires)
+
+        # use the token with latest approval expires date
+        new = old_tokens[-1].refresh(**token)
+
+        for old in old_tokens:
+            self.db.session.delete(old)
```

Comment:
This is cheap, but it works for me.  If you don't mind adding a comment with explanation of the error this prevents + link to https://github.com/sync-for-science/auth-proxy/issues/45, I think this is good to merge!